### DISCUSSION TITLE
How to set axis text alignment & text size?

### DISCUSSION THREAD

**Discussion by jnayee** [Live-Charts/LiveCharts2#176] [Q&A]
Hi! I would like to align Y-axis text to the right (currently set to centre by default) and adjust font size, how can I do this? 
![textalign](https://user-images.githubusercontent.com/89976865/132045414-e6fc76e6-54c9-4a1e-bd54-c7711843470b.png)

**Comment by alex4998** [comment]
Same question here. Especially curious how to change the size.

**Comment by jnayee** [comment]
Turns out there is a TextSize property for Axis, see example below:

` public IEnumerable<ICartesianAxis> YAxes { get; set; }
        = new Axis[]
        {
            new Axis
            {
                MinLimit = 0,
                MaxLimit = 10,
                ForceStepToMin = true,
                MinStep = 1,
                TextSize = 14,
                SeparatorsPaint = new SolidColorPaint
                {
                    Color = SKColors.Gray,
                    StrokeThickness = 2,
                    PathEffect = new DashEffect(new float[] { 3, 3 })
                }
            }
        };`

The full example can be found here:
https://github.com/beto-rodriguez/LiveCharts2/blob/master/samples/ViewModelsSamples/Axes/Style/ViewModel.cs


I'm still not sure how to set the text alignment.

**Comment by alex4998** [comment]
Finally, I thought I had to take a closer look at the source code. It looks like alignment is possible. There are HorizontalAlign and VerticalAlign properties on the LabelGeometry that is used in the Axis class but the default values are Align.Middle

        public Align VerticalAlign { get; set; } = Align.Middle;
        public Align HorizontalAlign { get; set; } = Align.Middle;

and there is no implementation to set them in the Axis. However I guess they can be changed if you get labels Axis.LabelsPaint and then get geometries from LabelsPaint.GetGeometries(), then cast items to LabelGeometry and change the alignment. It might be necessary to reinitialize something after that. Anyway, all that are just guesses.

I don't really need to change the alignment but if I do I would try to change the code, add new public properties to the Axis for alignment and use those properties values where AddGeometryToPaintTask is. Again, if I understand the logic right.

**Comment by jnayee** [comment]
Thank you for the guidance @alex4998, I will try to figure out the implementation as soon as possible. For now, to make the alignment less noticeable, I have applied String Format on the values using Labeler property as shown below:

`public IEnumerable<ICartesianAxis> YAxes { get; set; } = new Axis[] { Labeler = value => string.Format("{0:0.000}", value), ShowSeparatorLines = false, TextSize = 12 };`

<img src="https://user-images.githubusercontent.com/89976865/148652214-068d1de2-4331-4a9c-aac8-ff19fc4a1084.png"  height="300">
